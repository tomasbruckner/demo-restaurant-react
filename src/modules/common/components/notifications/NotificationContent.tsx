import Button from 'modules/common/components/controls/Button';
import SnackbarContent from '@material-ui/core/SnackbarContent';
import { createStyles, Theme, WithStyles, withStyles } from '@material-ui/core';
import React, { PureComponent } from 'react';
import styled from 'styled-components/macro';
import green from '@material-ui/core/colors/green';
import amber from '@material-ui/core/colors/amber';
import { MdCheckCircle, MdClose, MdError, MdInfo, MdWarning } from 'react-icons/md';

const variantIcon = {
  success: MdCheckCircle,
  warning: MdWarning,
  error: MdError,
  info: MdInfo,
};

const notificationStyles = (theme: Theme) =>
  createStyles({
    success: {
      backgroundColor: green[600],
    },
    error: {
      backgroundColor: theme.palette.error.dark,
    },
    info: {
      backgroundColor: theme.palette.primary.dark,
    },
    warning: {
      backgroundColor: amber[700],
    },
  });

const NotificationIcon = styled.span`
  margin-right: 8px;
  font-size: 20px;
  line-height: 20px;
  height: 20px;
  opacity: 0.9;
`;

const NotificationMessage = styled.span`
  display: flex;
  align-items: center;
`;

type Props = {
  message: string;
  variant: 'success' | 'error';
  onClose?: () => any;
} & WithStyles<typeof notificationStyles>;

export class NotificationContent extends PureComponent<Props> {
  render() {
    const { classes, message, onClose, variant } = this.props;

    const Icon = variantIcon[variant];

    const messageComponent = (
      <NotificationMessage id="idNotification">
        <NotificationIcon>
          <Icon />
        </NotificationIcon>
        {message}
      </NotificationMessage>
    );

    const actionComponent = (
      <Button asIcon key="close" aria-label="Close" color="inherit" onClick={onClose}>
        <MdClose />
      </Button>
    );

    return (
      <SnackbarContent
        className={classes[variant]}
        aria-describedby="idNotification"
        message={messageComponent}
        action={actionComponent}
      />
    );
  }
}

export default withStyles(notificationStyles)(NotificationContent);
