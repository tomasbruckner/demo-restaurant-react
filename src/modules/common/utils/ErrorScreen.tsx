import React, { PureComponent } from 'react';
import PageLayout from '../layouts/PageLayout';
import { I18N_CORE } from 'core/i18n/i18n';
import { WithTranslation, withTranslation } from 'react-i18next';

export class ErrorScreen extends PureComponent<{ title: string } & WithTranslation> {
  render() {
    const { t } = this.props;

    return <PageLayout title={t('page.error.title')} />;
  }
}

export default withTranslation(I18N_CORE)(ErrorScreen);
