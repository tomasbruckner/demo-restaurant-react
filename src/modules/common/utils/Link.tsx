import { Link as RouterLink } from 'react-router-dom';

export default RouterLink;
