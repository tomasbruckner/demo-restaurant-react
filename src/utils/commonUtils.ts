import format from 'date-fns/format';
import parseISO from 'date-fns/parseISO';

export const isRunningInJest = () => {
  return process.env.JEST_WORKER_ID !== undefined;
};

export const formatDate = (date: string) => {
  return date ? format(parseISO(date), 'MMM d, yyyy') : date;
};
