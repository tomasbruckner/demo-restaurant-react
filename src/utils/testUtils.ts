export const fakeTranslate = (key: string) => key;

export const fakeLoad = () => Promise.resolve();

export const fakeLogger = {
  info() {},
  error() {},
};

// eslint-disable-next-line no-unused-vars
export const fakeFunction = (): any => {};

export const fakeHistory = {
  push: () => {},
};

export const fakeTranslationProps = {
  t: fakeTranslate,
  i18n: {} as any,
  tReady: true,
};

export const fakeReview = {
  reviewId: 1,
  reviewRatingId: 2,
  restaurantId: 3,
  title: 'title',
  text: 'text',
  dateOfVisit: '',
  hasOwnerResponse: true,
  created: '',
  owner: {
    userId: 4,
    firstName: 'firstName',
    lastName: 'lastName',
    imageLink: 'imageLink',
  },
  ownerResponse: {
    reviewReplyId: 5,
    reviewId: 6,
    creatorId: 7,
    created: '',
    updated: '',
    text: 'text',
  },
  restaurantName: 'restaurantName',
};

export const fakeUser = {
  userId: 1,
  roleId: 2,
  firstName: 'firstName',
  lastName: 'lastName',
  email: 'email@email',
  created: '',
  imageLink: 'string',
};

export const fakeRestaurant = {
  restaurantId: 1,
  ownerId: 2,
  name: 'name',
  description: 'description',
  websiteUrl: 'websiteUrl',
  contactPhone: 'contactPhone',
  address: 'address',
  numberOfReviews: 3,
  averageRating: 4,
  hasCurrentUserReviewed: true,
  imageLink: 'imageLink',
};

export const fakeRouterProps = {
  location: {} as any,
  match: {} as any,
  history: { push: () => {} } as any,
};
