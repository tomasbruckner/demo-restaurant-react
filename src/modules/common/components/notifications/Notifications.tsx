import Snackbar from '@material-ui/core/Snackbar';
import { getNotificationSelector, removeNotification } from 'core/redux/ducks/NotificationDuck';
import React, { PureComponent } from 'react';
import { connect, ConnectedProps } from 'react-redux';
import NotificationContent from './NotificationContent';
import { RootState } from 'core/redux/store';

const NOTIFICATION_TIMEOUT = 5000;

const mapStateToProps = (state: RootState) => {
  return {
    notification: getNotificationSelector(state),
  };
};

const mapDispatchToProps = {
  removeNotificationAction: removeNotification,
};
const connector = connect(mapStateToProps, mapDispatchToProps);

export class Notifications extends PureComponent<ConnectedProps<typeof connector>> {
  notificationTimeoutId?: number = undefined;

  componentDidUpdate() {
    const { notification, removeNotificationAction } = this.props;

    if (notification) {
      clearTimeout(this.notificationTimeoutId);
      this.notificationTimeoutId = setTimeout(removeNotificationAction, NOTIFICATION_TIMEOUT);
    }
  }

  render() {
    const { notification, removeNotificationAction } = this.props;

    if (!notification) {
      return null;
    }

    const { message, variant } = notification;

    return (
      <Snackbar
        anchorOrigin={{
          vertical: 'bottom',
          horizontal: 'right',
        }}
        open={!!notification}
      >
        <NotificationContent
          onClose={removeNotificationAction}
          variant={variant}
          message={message}
        />
      </Snackbar>
    );
  }
}

export default connector(Notifications);
