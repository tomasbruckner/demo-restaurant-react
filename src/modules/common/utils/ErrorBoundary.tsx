import React, { PureComponent } from 'react';
import { WithTranslation, withTranslation } from 'react-i18next';
import Logger from 'utils/logger';
import { I18N_CORE } from 'core/i18n/i18n';
import ErrorScreen from './ErrorScreen';

export class ErrorBoundaryComp extends PureComponent<
  { renderProp?: () => React.ReactNode } & WithTranslation
> {
  state = {
    error: null,
  };

  static getDerivedStateFromError(error: Error) {
    return { error };
  }

  componentDidCatch(error: unknown, errorInfo: unknown) {
    Logger.error(error, errorInfo);
  }

  render() {
    const { children, renderProp, t } = this.props;
    const { error } = this.state;

    if (!error) {
      return children;
    }

    if (renderProp) {
      return renderProp();
    }

    return <ErrorScreen title={t('CommonComponents.ErrorBoundary.error')} />;
  }
}

export default withTranslation(I18N_CORE)(ErrorBoundaryComp);
