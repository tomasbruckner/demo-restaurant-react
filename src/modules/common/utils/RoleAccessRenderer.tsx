import React, { PureComponent, ReactNode } from 'react';
import { connect, ConnectedProps } from 'react-redux';
import { RootState } from 'core/redux/store';
import { getUserFromState } from 'core/redux/ducks/UserDuck';
import { Role } from 'utils/constants';

const connector = connect((state: RootState) => {
  return {
    user: getUserFromState(state),
  };
});

type Props = {
  onlyAdmin?: boolean;
  adminOrOwner?: boolean;
  renderNotAllowed?: ReactNode;
  children: React.ReactNode;
};

class RoleAccessRenderer extends PureComponent<Props & ConnectedProps<typeof connector>> {
  checkPermissions = () => {
    const { user, onlyAdmin, adminOrOwner } = this.props;

    if (!user) {
      return false;
    }

    if (onlyAdmin) {
      return user.roleId === Role.Admin;
    }

    if (adminOrOwner) {
      return user.roleId === Role.Admin || user.roleId === Role.Owner;
    }

    return false;
  };

  render() {
    const { children, renderNotAllowed = null } = this.props;

    if (this.checkPermissions()) {
      return children;
    }

    return renderNotAllowed;
  }
}

export default connector(RoleAccessRenderer);
