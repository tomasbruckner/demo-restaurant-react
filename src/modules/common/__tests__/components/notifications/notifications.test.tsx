import React from 'react';
import 'jest-styled-components';
import { Notifications } from 'modules/common/components/notifications/Notifications';
import { fakeFunction, fakeTranslationProps } from 'utils/testUtils';
import { ActionCreatorWithoutPayload } from '@reduxjs/toolkit';
import EnzymeToJson from 'enzyme-to-json';
import { mount } from 'enzyme';

describe('Notifications', () => {
  it('basic example', () => {
    const tree = EnzymeToJson(
      mount(
        <Notifications
          removeNotificationAction={fakeFunction as ActionCreatorWithoutPayload}
          notification={{ message: 'notif', variant: 'success' }}
          {...fakeTranslationProps}
        />,
      ),
    );

    expect(tree).toMatchSnapshot();
  });
});
