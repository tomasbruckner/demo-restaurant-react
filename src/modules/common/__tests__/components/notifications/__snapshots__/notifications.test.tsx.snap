// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Notifications basic example 1`] = `
.c1 {
  margin-right: 8px;
  font-size: 20px;
  line-height: 20px;
  height: 20px;
  opacity: 0.9;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

<Notifications
  i18n={Object {}}
  notification={
    Object {
      "message": "notif",
      "variant": "success",
    }
  }
  removeNotificationAction={[Function]}
  t={[Function]}
  tReady={true}
>
  <WithStyles(ForwardRef(Snackbar))
    anchorOrigin={
      Object {
        "horizontal": "right",
        "vertical": "bottom",
      }
    }
    open={true}
  >
    <ForwardRef(Snackbar)
      anchorOrigin={
        Object {
          "horizontal": "right",
          "vertical": "bottom",
        }
      }
      classes={
        Object {
          "anchorOriginBottomCenter": "MuiSnackbar-anchorOriginBottomCenter",
          "anchorOriginBottomLeft": "MuiSnackbar-anchorOriginBottomLeft",
          "anchorOriginBottomRight": "MuiSnackbar-anchorOriginBottomRight",
          "anchorOriginTopCenter": "MuiSnackbar-anchorOriginTopCenter",
          "anchorOriginTopLeft": "MuiSnackbar-anchorOriginTopLeft",
          "anchorOriginTopRight": "MuiSnackbar-anchorOriginTopRight",
          "root": "MuiSnackbar-root",
        }
      }
      open={true}
    >
      <ClickAwayListener
        onClickAway={[Function]}
      >
        <div
          className="MuiSnackbar-root MuiSnackbar-anchorOriginBottomRight"
          onClick={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
          onTouchEnd={[Function]}
        >
          <ForwardRef(Grow)
            appear={true}
            direction="up"
            in={true}
            onEnter={[Function]}
            onExited={[Function]}
            timeout={
              Object {
                "enter": 225,
                "exit": 195,
              }
            }
          >
            <Transition
              addEndListener={[Function]}
              appear={true}
              direction="up"
              enter={true}
              exit={true}
              in={true}
              mountOnEnter={false}
              onEnter={[Function]}
              onEntered={[Function]}
              onEntering={[Function]}
              onExit={[Function]}
              onExited={[Function]}
              onExiting={[Function]}
              timeout={
                Object {
                  "enter": 225,
                  "exit": 195,
                }
              }
              unmountOnExit={false}
            >
              <WithStyles(NotificationContent)
                direction="up"
                message="notif"
                onClose={[Function]}
                style={
                  Object {
                    "opacity": 1,
                    "transform": "scale(1, 1)",
                    "visibility": undefined,
                  }
                }
                variant="success"
              >
                <NotificationContent
                  classes={
                    Object {
                      "error": "NotificationContent-error-2",
                      "info": "NotificationContent-info-3",
                      "success": "NotificationContent-success-1",
                      "warning": "NotificationContent-warning-4",
                    }
                  }
                  direction="up"
                  message="notif"
                  onClose={[Function]}
                  style={
                    Object {
                      "opacity": 1,
                      "transform": "scale(1, 1)",
                      "visibility": undefined,
                    }
                  }
                  variant="success"
                >
                  <WithStyles(ForwardRef(SnackbarContent))
                    action={
                      <ButtonComponent
                        aria-label="Close"
                        asIcon={true}
                        color="inherit"
                        onClick={[Function]}
                      >
                        <MdClose />
                      </ButtonComponent>
                    }
                    aria-describedby="idNotification"
                    className="NotificationContent-success-1"
                    message={
                      <ForwardRef(NotificationContent__NotificationMessage)
                        id="idNotification"
                      >
                        <ForwardRef(NotificationContent__NotificationIcon)>
                          <MdCheckCircle />
                        </ForwardRef(NotificationContent__NotificationIcon)>
                        notif
                      </ForwardRef(NotificationContent__NotificationMessage)>
                    }
                  >
                    <ForwardRef(SnackbarContent)
                      action={
                        <ButtonComponent
                          aria-label="Close"
                          asIcon={true}
                          color="inherit"
                          onClick={[Function]}
                        >
                          <MdClose />
                        </ButtonComponent>
                      }
                      aria-describedby="idNotification"
                      className="NotificationContent-success-1"
                      classes={
                        Object {
                          "action": "MuiSnackbarContent-action",
                          "message": "MuiSnackbarContent-message",
                          "root": "MuiSnackbarContent-root",
                        }
                      }
                      message={
                        <ForwardRef(NotificationContent__NotificationMessage)
                          id="idNotification"
                        >
                          <ForwardRef(NotificationContent__NotificationIcon)>
                            <MdCheckCircle />
                          </ForwardRef(NotificationContent__NotificationIcon)>
                          notif
                        </ForwardRef(NotificationContent__NotificationMessage)>
                      }
                    >
                      <WithStyles(ForwardRef(Paper))
                        aria-describedby="idNotification"
                        className="MuiSnackbarContent-root NotificationContent-success-1"
                        elevation={6}
                        role="alert"
                        square={true}
                      >
                        <ForwardRef(Paper)
                          aria-describedby="idNotification"
                          className="MuiSnackbarContent-root NotificationContent-success-1"
                          classes={
                            Object {
                              "elevation0": "MuiPaper-elevation0",
                              "elevation1": "MuiPaper-elevation1",
                              "elevation10": "MuiPaper-elevation10",
                              "elevation11": "MuiPaper-elevation11",
                              "elevation12": "MuiPaper-elevation12",
                              "elevation13": "MuiPaper-elevation13",
                              "elevation14": "MuiPaper-elevation14",
                              "elevation15": "MuiPaper-elevation15",
                              "elevation16": "MuiPaper-elevation16",
                              "elevation17": "MuiPaper-elevation17",
                              "elevation18": "MuiPaper-elevation18",
                              "elevation19": "MuiPaper-elevation19",
                              "elevation2": "MuiPaper-elevation2",
                              "elevation20": "MuiPaper-elevation20",
                              "elevation21": "MuiPaper-elevation21",
                              "elevation22": "MuiPaper-elevation22",
                              "elevation23": "MuiPaper-elevation23",
                              "elevation24": "MuiPaper-elevation24",
                              "elevation3": "MuiPaper-elevation3",
                              "elevation4": "MuiPaper-elevation4",
                              "elevation5": "MuiPaper-elevation5",
                              "elevation6": "MuiPaper-elevation6",
                              "elevation7": "MuiPaper-elevation7",
                              "elevation8": "MuiPaper-elevation8",
                              "elevation9": "MuiPaper-elevation9",
                              "outlined": "MuiPaper-outlined",
                              "root": "MuiPaper-root",
                              "rounded": "MuiPaper-rounded",
                            }
                          }
                          elevation={6}
                          role="alert"
                          square={true}
                        >
                          <div
                            aria-describedby="idNotification"
                            className="MuiPaper-root MuiSnackbarContent-root NotificationContent-success-1 MuiPaper-elevation6"
                            role="alert"
                          >
                            <div
                              className="MuiSnackbarContent-message"
                            >
                              <NotificationContent__NotificationMessage
                                id="idNotification"
                              >
                                <span
                                  className="c0"
                                  id="idNotification"
                                >
                                  <NotificationContent__NotificationIcon>
                                    <span
                                      className="c1"
                                    >
                                      <MdCheckCircle>
                                        <IconBase
                                          attr={
                                            Object {
                                              "viewBox": "0 0 24 24",
                                            }
                                          }
                                        >
                                          <svg
                                            fill="currentColor"
                                            height="1em"
                                            stroke="currentColor"
                                            strokeWidth="0"
                                            style={
                                              Object {
                                                "color": undefined,
                                              }
                                            }
                                            viewBox="0 0 24 24"
                                            width="1em"
                                            xmlns="http://www.w3.org/2000/svg"
                                          >
                                            <path
                                              d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm-2 15l-5-5 1.41-1.41L10 14.17l7.59-7.59L19 8l-9 9z"
                                              key="0"
                                            />
                                          </svg>
                                        </IconBase>
                                      </MdCheckCircle>
                                    </span>
                                  </NotificationContent__NotificationIcon>
                                  notif
                                </span>
                              </NotificationContent__NotificationMessage>
                            </div>
                            <div
                              className="MuiSnackbarContent-action"
                            >
                              <ButtonComponent
                                aria-label="Close"
                                asIcon={true}
                                color="inherit"
                                key="close"
                                onClick={[Function]}
                              >
                                <WithStyles(ForwardRef(IconButton))
                                  aria-label="Close"
                                  color="inherit"
                                  onClick={[Function]}
                                >
                                  <ForwardRef(IconButton)
                                    aria-label="Close"
                                    classes={
                                      Object {
                                        "colorInherit": "MuiIconButton-colorInherit",
                                        "colorPrimary": "MuiIconButton-colorPrimary",
                                        "colorSecondary": "MuiIconButton-colorSecondary",
                                        "disabled": "Mui-disabled",
                                        "edgeEnd": "MuiIconButton-edgeEnd",
                                        "edgeStart": "MuiIconButton-edgeStart",
                                        "label": "MuiIconButton-label",
                                        "root": "MuiIconButton-root",
                                        "sizeSmall": "MuiIconButton-sizeSmall",
                                      }
                                    }
                                    color="inherit"
                                    onClick={[Function]}
                                  >
                                    <WithStyles(ForwardRef(ButtonBase))
                                      aria-label="Close"
                                      centerRipple={true}
                                      className="MuiIconButton-root MuiIconButton-colorInherit"
                                      disabled={false}
                                      focusRipple={true}
                                      onClick={[Function]}
                                    >
                                      <ForwardRef(ButtonBase)
                                        aria-label="Close"
                                        centerRipple={true}
                                        className="MuiIconButton-root MuiIconButton-colorInherit"
                                        classes={
                                          Object {
                                            "disabled": "Mui-disabled",
                                            "focusVisible": "Mui-focusVisible",
                                            "root": "MuiButtonBase-root",
                                          }
                                        }
                                        disabled={false}
                                        focusRipple={true}
                                        onClick={[Function]}
                                      >
                                        <button
                                          aria-label="Close"
                                          className="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorInherit"
                                          disabled={false}
                                          onBlur={[Function]}
                                          onClick={[Function]}
                                          onDragLeave={[Function]}
                                          onFocus={[Function]}
                                          onKeyDown={[Function]}
                                          onKeyUp={[Function]}
                                          onMouseDown={[Function]}
                                          onMouseLeave={[Function]}
                                          onMouseUp={[Function]}
                                          onTouchEnd={[Function]}
                                          onTouchMove={[Function]}
                                          onTouchStart={[Function]}
                                          tabIndex={0}
                                          type="button"
                                        >
                                          <span
                                            className="MuiIconButton-label"
                                          >
                                            <MdClose>
                                              <IconBase
                                                attr={
                                                  Object {
                                                    "viewBox": "0 0 24 24",
                                                  }
                                                }
                                              >
                                                <svg
                                                  fill="currentColor"
                                                  height="1em"
                                                  stroke="currentColor"
                                                  strokeWidth="0"
                                                  style={
                                                    Object {
                                                      "color": undefined,
                                                    }
                                                  }
                                                  viewBox="0 0 24 24"
                                                  width="1em"
                                                  xmlns="http://www.w3.org/2000/svg"
                                                >
                                                  <path
                                                    d="M19 6.41L17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"
                                                    key="0"
                                                  />
                                                </svg>
                                              </IconBase>
                                            </MdClose>
                                          </span>
                                          <WithStyles(memo)
                                            center={true}
                                          >
                                            <ForwardRef(TouchRipple)
                                              center={true}
                                              classes={
                                                Object {
                                                  "child": "MuiTouchRipple-child",
                                                  "childLeaving": "MuiTouchRipple-childLeaving",
                                                  "childPulsate": "MuiTouchRipple-childPulsate",
                                                  "ripple": "MuiTouchRipple-ripple",
                                                  "ripplePulsate": "MuiTouchRipple-ripplePulsate",
                                                  "rippleVisible": "MuiTouchRipple-rippleVisible",
                                                  "root": "MuiTouchRipple-root",
                                                }
                                              }
                                            >
                                              <span
                                                className="MuiTouchRipple-root"
                                              >
                                                <TransitionGroup
                                                  childFactory={[Function]}
                                                  component={null}
                                                  exit={true}
                                                />
                                              </span>
                                            </ForwardRef(TouchRipple)>
                                          </WithStyles(memo)>
                                        </button>
                                      </ForwardRef(ButtonBase)>
                                    </WithStyles(ForwardRef(ButtonBase))>
                                  </ForwardRef(IconButton)>
                                </WithStyles(ForwardRef(IconButton))>
                              </ButtonComponent>
                            </div>
                          </div>
                        </ForwardRef(Paper)>
                      </WithStyles(ForwardRef(Paper))>
                    </ForwardRef(SnackbarContent)>
                  </WithStyles(ForwardRef(SnackbarContent))>
                </NotificationContent>
              </WithStyles(NotificationContent)>
            </Transition>
          </ForwardRef(Grow)>
        </div>
      </ClickAwayListener>
    </ForwardRef(Snackbar)>
  </WithStyles(ForwardRef(Snackbar))>
</Notifications>
`;
